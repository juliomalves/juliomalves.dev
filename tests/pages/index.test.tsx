import * as React from 'react'
import { render, screen } from '@testing-library/preact'
import HomePage from '@/pages/index'

const renderComponent = () => render(<HomePage />)

describe('GIVEN a <HomePage />', () => {
    it('THEN should render its children components', () => {
        renderComponent()
        expect(screen.getByRole('main')).toBeInTheDocument()
        expect(screen.getByLabelText('Summary')).toBeInTheDocument()
        expect(screen.getByLabelText('Tech Skills')).toBeInTheDocument()
        expect(screen.getByLabelText('Timeline')).toBeInTheDocument()
        expect(screen.getByLabelText('Contact')).toBeInTheDocument()
    })
})
