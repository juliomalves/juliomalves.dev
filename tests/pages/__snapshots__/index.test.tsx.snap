// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="py-20"
  >
    <h1
      class="text-5xl text-center text-accent-1"
    >
      Next.js + TypeScript + TailwindCSS
    </h1>
  </div>
</DocumentFragment>
`;
